### DISCUSSION TITLE
JSON Objekt als Schalter nutzen

### DISCUSSION THREAD

**Discussion by mcuiobroker** [Zefau/ioBroker.jarvis#1826] [Q&A]
In einem ioBroker DP kommt ein JSON von MQTT.
an -> {"STATE":"ON"}
aus -> {"STATE":"OFF"}
Wie kann man das realisieren?

Versuche
Datenpunkt-Eigenschaften
`{"value":"val => JSON.parse(val).STATE","setOn":"{\"STATE\":\"ON\"}","setOff":"{\"STATE\":\"OFF\"}"}`
oder
`{"value":"val => JSON.parse(val).STATE","on":"{\"STATE\":\"ON\"}","off":"{\"STATE\":\"OFF\"}"}`
![image](https://user-images.githubusercontent.com/84818934/186965768-a920dda2-295c-4ca9-960d-200e3784ca3a.png)

`{"value":"val => JSON.parse(val).STATE","on":"{'STATE':'ON'}","off":"{'STATE':'OFF'}"}`

![image](https://user-images.githubusercontent.com/84818934/186965337-53ca4698-0e57-4f19-a34f-f45ab27497c1.png)
![image](https://user-images.githubusercontent.com/84818934/186965380-2cf67326-846a-4f1a-8d34-5a82331a095f.png)
Meldung
![image](https://user-images.githubusercontent.com/84818934/186965434-230fdee4-09eb-4f06-bd5b-4969d7e4b3b5.png)

**Comment by Zefau** [comment]
Hast du das lösen können? 
```
 {"value":"val => JSON.parse(val).STATE","setOn":"{\"STATE\":\"ON\"}","setOff":"{\"STATE\":\"OFF\"}"}
```
sieht eigl. korrekt aus. Ggf. auch `{` und `}` escapen?

**Reply by mcuiobroker** [reply]
Nein.
